[PoC] Document DevOps proposed workarounds (#210)
open-telemetry/opentelemetry-dotnet-instrumentation: README.md

```sh
sampleAppTargetFramework=net472 sampleApp=BindingRedirect ./poc.sh
```

> ## Handling of Assembly version Conflicts

Reviewer: shouldn't this section live in  [docs/USAGE.md#troubleshooting](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/poc-otel-sdk/docs/USAGE.md#troubleshooting) ?

> OpenTelemetry SDK NuGet package deployed with the OpenTelemetry .NET Instrumentation.

Reviewer: ```suggestion
OpenTelemetry SDK NuGet packages are deployed with the OpenTelemetry .NET Instrumentation.
```

> Conflicts in assemblies referenced by "source instrumentations," should be handled in

Reviewer: ```suggestion
Conflicts in assemblies referenced by "source instrumentations", should be handled in
```

the same way: updating the project references to ensure that they are the same version
as the one used by the instrumentation.

However, the workarounds proposed above only work at build time. When a rebuild is not
possible, there are ways to force the application to use the assembly versions shipped
together with the instrumentation.

For .NET Framework applications, the workaround is to use Binding Redirects. For .NET Core

> applications, the workaround is to manipulate the deps.json file. For the PoC, we are not
> planning to automate any of these workarounds, bur we are manually validating them.

Reviewer: ```suggestion
applications, the workaround is to manipulate the `deps.json`. Currently, we do not
automate any of these workarounds, but we are manually validating them.
```
